localstack/services/cloudwatch/cloudwatch_database_helper.py

```diff
@@ -0,0 +1,407 @@
+
+                elif stat == "Average":
+                    total_sum = 0
+                    count = 0
+                    for cleaned_value in cleaned_values:
+                        if isinstance(cleaned_value, tuple):
+                            total_sum += cleaned_value[0]
+                            count += cleaned_value[1]
+                        else:
+                            total_sum += cleaned_value
+                            count += 1
+                    cleaned_datapoints[timestamp] = total_sum / count
+
+            return {
+                "id": query.get("Id"),
+                "datapoints": cleaned_datapoints,
+                "label": f"{metric.get('MetricName')} {stat}",
+            }
+
+    def list_metrics(self, account_id, region, namespace, metric_name, dimensions) -> dict:
+        with sqlite3.connect(self.METRICS_DB) as conn:
+            cur = conn.cursor()
+
+            data = (account_id, region)
+
+            namespace_filter = ""
+            if namespace:
+                namespace_filter = "AND namespace = ?"
+                data = data + (namespace,)
+
+            metric_name_filter = ""
+            if metric_name:
+                metric_name_filter = "AND metric_name = ?"
+                data = data + (metric_name,)
+
+            dimension_filter = ""
+            for dimension in dimensions:
+                dimension_filter += "AND dimensions LIKE ? "
+                data = data + (f"%{dimension.get('Name')}={dimension.get('Value','')}%",)
+
+            query = f"""
+                SELECT DISTINCT metric_name, namespace, dimensions
+                FROM {self.TABLE_SINGLE_METRICS}
+                WHERE account_id = ? AND region = ? {namespace_filter} {metric_name_filter} {dimension_filter}
+                ORDER BY timestamp DESC
+            """
+
+            cur.execute(
+                query,
+                data,
+            )
+            single_metrics_result = [
+                {
+                    "metric_name": r[0],
+                    "namespace": r[1],
+                    "dimensions": self._restore_dimensions_from_string(r[2]),
+                }
+                for r in cur.fetchall()
+            ]
+
+            query = f"""
+                SELECT DISTINCT metric_name, namespace ,dimensions FROM {self.TABLE_AGGREGATED_METRICS}
+                WHERE account_id = ? AND region = ?
+                {namespace_filter}
+                {metric_name_filter}
+                {dimension_filter}
+                ORDER BY timestamp DESC
+            """
+
+            cur.execute(
+                query,
+                data,
+            )
+            aggregated_metrics_result = [
+                {
+                    "metric_name": r[0],
+                    "namespace": r[1],
+                    "dimensions": self._restore_dimensions_from_string(r[2]),
+                }
+                for r in cur.fetchall()
+            ]
```

Comment:
alternatively we could use just one combined query, simplified (need to add `WHERE` clauses still):

```
SELECT DISTINCT metric_name, namespace, dimensions
FROM (
    SELECT metric_name, namespace, dimensions
    FROM SINGLE_METRICS
    UNION
    SELECT metric_name, namespace, dimensions
    FROM AGGREGATED_METRICS
) AS combined
```